feat: injectTypes docs (#9012)
withastro/docs: src/content/docs/en/reference/integrations-reference.mdx

A read-only copy of the user-supplied [Astro config](/en/reference/configuration-reference/). This is resolved _after_ other integrations have run.

> #### `setAdapter` option

Reviewer: Just checking that this isn't new (4.14) but was previously undocumented, so it doesn't need a `<Since />`?

Author: Yes exactly. It was only mentioned in the quick reference and thought it wouldn't hurt to add it

**Type:** `(adapter: AstroAdapter) => void;`

Makes the integration an adapter. Read more in the [adapter API](/en/reference/adapter-reference/).

#### `injectTypes` options

<p><Since v="4.14.0" /></p>

**Type:** `(injectedType: { filename: string; content: string }) => URL`

> Allows you to inject types into your user's project. `filename` must ends with `".d.ts"` and `content` must be valid TypeScript.

Reviewer: ```suggestion
Allows you to inject types into your user's project by adding a new a `*.d.ts` file.

The `filename` property will be used to generate a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and must end with `".d.ts"` 

The `content` property will create the body of the file, and must be valid TypeScript.
```

> Allows you to inject types easily in your user's project. `filename` must ends with `".d.ts"` and `content` must be valid TypeScript (it will be formatted).

Reviewer: ```suggestion
Allows you to inject types into your user's project. `filename` must ends with `".d.ts"` and `content` must be valid TypeScript.
```
- I'm not sure "in" vs "into" because it depends on what is actually happening here. But, if "into" doesn't sound "wrong" to you, then I suspect it's the correct one.

- The "it will be formatted for you" feels a *little* at odds with saying what you DO have to do. Does this suffer too much if we simply don't say this?

Author: - in vs into: it does a bunch of wiring under the hood so not sure what to use either! I think you know the nuance better so I trust you!
- It's not formatted anymore actually, the PR was updated

> Under the hood, it will create a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and create references to it. `injectTypes` returns a URL to the normalized path.

Reviewer: ```suggestion
 This will create a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and create references to it. `injectTypes` returns a URL to the normalized path.
```
I would put this BELOW the code example, since we've just described what the `filename` and `content` should look like in the example above. Probably a good time to let them see it in action right here, THEN explain what this resulting code will do.

Author: There was so much to explain in there, I was not sure how to organize it! Whatever you prefer

A read-only copy of the user-supplied [Astro config](/en/reference/configuration-reference/). This is resolved _after_ other integrations have run.

#### `setAdapter` option

**Type:** `(adapter: AstroAdapter) => void;`

Makes the integration an adapter. Read more in the [adapter API](/en/reference/adapter-reference/).

#### `injectTypes` options

<p><Since v="4.14.0" /></p>

**Type:** `(injectedType: { filename: string; content: string }) => URL`

Allows you to inject types into your user's project by adding a new a `*.d.ts` file.

> The `filename` property will be used to generate a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and must end with `".d.ts"` 

Author: ```suggestion
The `filename` property will be used to generate a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and must end with `".d.ts"`.
```

The `content` property will create the body of the file, and must be valid TypeScript.

> Additionally, `injectTypes()` returns a URL to the normalized path so you can/to allow for...

Author: ```suggestion
Additionally, `injectTypes()` returns a URL to the normalized path so you can overwrite its content later on, or manipulate it in any way you want.
```

>  This will create a file at `/.astro/integrations/<normalized_integration_name>/<normalized_filename>.d.ts` and create references to it. `injectTypes` returns a URL to the normalized path.

Reviewer: ```suggestion
Additionally, `injectTypes()` returns a URL to the normalized path so you can/to allow for...
```
